### ISSUE TITLE
Run-Time check Failure #3 when building debug build in VS 15.8.8

### ISSUE THREAD

**Issue by mgejke** [simbody/simbody#648]
After upgrading to the latest VS we have begun experiencing exceptions when running
`AssemblySolver::track`

`Run-Time Check Failure #3 - The variable 'iback' is being used without being initialized.`

The exception is thrown in **lbfgsb.cpp**, in the function **mainlb_**(and rightfully so, the variables are uninitialized, although I think it's fine, it's only storing the local state of the function).

Initializing all the variables to zero at the beginning of the function fixes the issue and seem to give the same results as before(before upgrading VS and Release-build). 

Would that be considered a proper solution?

**Comment by sherm1** [comment]
That sounds reasonable but please investigate a little further. There are lots of variables in that ugly function -- it wasn't obvious to me why `iback` is different than any of the others. Possibly one of the called functions that is passed `&iback` was expected to set it but didn't? Did the error message specify a particular line number? I wonder if you would get more information in a Debug build.

Because this is inscrutable machine-translated code I would suggest doing the minimal change possible (after investigating) unless you are very sure you know what's happening. If it is just a problem with iback, maybe just initialize it.

**Comment by mgejke** [comment]
It's not just `iback`,  the following lines give the exception if I continue debugging.

```
4864    isave[4] = iback;
4868    isave[8] = itail;
4874    isave[15] = ifun;
4875    isave[16] = iword;
4877    isave[18] = nact;
4878    isave[19] = ileave;
4879    isave[20] = nenter;
4881    dsave[2] = fold;
4883    dsave[4] = dnorm;
4885    dsave[6] = cpu1;
4889    dsave[10] = time1;
4890    dsave[11] = gd;
4891    dsave[12] = stpmx;
4892    dsave[13] = sbgnrm;
4894    dsave[15] = gdold;
4895    dsave[16] = dtd;
```

It's hard to downgrade VS2017 to a previous minor version to test properly, but upgrading is the only change that we made, and before that we had Debug-builds without the exceptions.

Initializing all the variables with this patch makes the exceptions go away. I guess the bigger question is if we're doing something wrong and that why the variables are uninitialized. 

```
diff --git a/SimTKmath/Optimizers/src/lbfgsb.cpp b/SimTKmath/Optimizers/src/lbfgsb.cpp
index c07e6960..300af867 100644
--- a/SimTKmath/Optimizers/src/lbfgsb.cpp
+++ b/SimTKmath/Optimizers/src/lbfgsb.cpp
@@ -4134,42 +4134,42 @@ actorization in formt;\002,/,\002   refresh the lbfgs memory and restart the\
  /*   int f_open(), s_wsfe(), do_fio(), e_wsfe(); */
 
     /* Local variables */
-    int head;
-    Real fold;
-    int nact;
-    Real ddum;
-    int info;
-    Real time;
-    int nfgv, ifun, iter, nint;
-    char word[5];
-    Real time1, time2;
-    int i__, iback, k;
-    Real gdold;
-    int nfree;
-    bool boxed;
-    int itail;
-    Real theta;
-    Real dnorm;
-    int nskip, iword;
-    Real xstep, stpmx;
-    Real gd, dr, rr;
-    int ileave;
-    int itfile;
-    Real cachyt, epsmch;
-    bool updatd;
-    Real sbtime;
-    bool prjctd;
-    int iupdat;
-    bool cnstnd;
-    Real sbgnrm;
-    int nenter;
-    Real lnscht;
-    int nintol;
-    Real dtd;
-    int col;
-    Real tol;
-    bool wrk;
-    Real stp, cpu1, cpu2;
+    int head = 0;
+    Real fold = 0;
+    int nact = 0;
+    Real ddum = 0;
+    int info = 0;
+    Real time = 0;
+    int nfgv = 0, ifun = 0, iter = 0, nint = 0;
+    char word[5]{0,0,0,0,0};
+    Real time1 = 0, time2 = 0;
+    int i__ = 0, iback = 0, k = 0;
+    Real gdold = 0;
+    int nfree = 0;
+    bool boxed = 0;
+    int itail = 0;
+    Real theta = 0;
+    Real dnorm = 0;
+    int nskip = 0, iword = 0;
+    Real xstep = 0, stpmx = 0;
+    Real gd = 0, dr = 0, rr = 0;
+    int ileave = 0;
+    int itfile = 0;
+    Real cachyt = 0, epsmch = 0;
+    bool updatd = 0;
+    Real sbtime = 0;
+    bool prjctd = 0;
+    int iupdat = 0;
+    bool cnstnd = 0;
+    Real sbgnrm = 0;
+    int nenter = 0;
+    Real lnscht = 0;
+    int nintol = 0;
+    Real dtd = 0;
+    int col = 0;
+    Real tol = 0;
+    bool wrk = 0;
+    Real stp = 0, cpu1 = 0, cpu2 = 0;
 
     /* Fortran I/O blocks */
 /*

```

**Comment by sherm1** [comment]
Thanks for investigating -- that looks like the right fix. Do you want to submit a PR to fix it?

**Comment by sherm1** [comment]
Resolved by PR #649 -- thanks, Martin!
